torch/testing/_internal/schema_check_tensor.py

```diff
@@ -38,16 +43,69 @@ def __repr__(self):
     @classmethod
     def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
         def unwrap(e):
-            return e.elem if isinstance(e, cls) else e
+            if isinstance(e, cls):
+                return e.elem
+            elif is_iterable_of_dtype(e, cls):
+                return [t.elem for t in e]
+            else:
+                return e
 
         def wrap(e):
-            return cls(e) if isinstance(e, torch.Tensor) else e
+            if isinstance(e, torch.Tensor):
+                return cls(e)
+            elif is_iterable_of_dtype(e, torch.Tensor):
+                return [cls(t) for t in e]
+            else:
+                return e
+
+        def has_mutated(before, after):
+            if isinstance(before, torch.Tensor) and isinstance(after, torch.Tensor):
+                return not torch.equal(before, after)
+            elif is_iterable_of_dtype(before, torch.Tensor) and is_iterable_of_dtype(after, torch.Tensor):
+                before_list = list(before)
+                after_list = list(after)
+                if (len(before_list) != len(after_list)):
+                    return True
+                for before_elem, after_elem in zip(before_list, after_list):
+                    if (not torch.equal(before_elem, after_elem)):
+                        return True
+                return False
+            else:
+                return False
 
         global schema_check_recorded_ops
         schema_check_recorded_ops.append(func.__name__)
+        schema, arguments = find_matching_schema(func, tree_map(unwrap, args), tree_map(unwrap, kwargs))
+        cloned_arguments = dict(zip(arguments.keys(), clone_inputs(arguments.values())))
         out = func(*tree_map(unwrap, args), **tree_map(unwrap, kwargs))
+
+        for argument in schema.arguments:
+            name = argument.name if argument.name != "self" else "input"
+            if (arguments.get(name) is not None and cloned_arguments.get(name) is not None):
```

Comment:
isn't it enough to check `arguments.get(name)`?
`cloned_arguments.get(name)` should be an assert?